data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeEvaluator.java

```diff
@@ -31,11 +31,11 @@ public ParseTreeEvaluator(final OperatorProvider operatorProvider, final ParseTr
     }
 
     @Override
-    public Boolean evaluate(ParseTree parseTree, Event event) {
+    public Object evaluate(ParseTree parseTree, Event event) {
         try {
             final ParseTreeEvaluatorListener listener = new ParseTreeEvaluatorListener(operatorProvider, coercionService, event);
             walker.walk(listener, parseTree);
-            return coercionService.coerce(listener.getResult(), Boolean.class);
+            return listener.getResult();
```

Comment:
Are there any possible limitations here? Could this break some existing scenarios?

Reply:
I can't think of cases where it can break the existing scenarios. All the existing scenarios are changed to use Boolean and they all check for return Object type to be Boolean and if not throw an error.